syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ResultCode
{
    uint32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest
{
    string name = 1;
    string pwd = 2;
}

message LoginResponse
{
    ResultCode result = 1;
    uint32 success = 2;
}

message RegisterRequest
{
    uint32 id = 1;
    string name = 2;
    string pwd = 3;
}

message RegisterResponse
{
    ResultCode result = 1;
    uint32 success = 2;
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}